### PR TITLE
EmbedSource component and package.json update

### PR DESCRIPTION
**Pull request by mcolsen**
# Description

Adds EmbedSource component that automatically embeds sources from Twitter, designed to be easy to add other embeddable source types later (I know there was some discussion of YouTube, for instance) 

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## Change Status

- [x] Complete, but not tested (may need new tests)

# How Has This Been Tested?

- [ ] Tests still need to be fixed 

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] There are no merge conflicts

### CODE REVIEW

File: src/components/EmbedSource.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import { useState, useEffect } from 'react';
+import { Spin } from 'antd';
+import { TwitterTweetEmbed } from 'react-twitter-embed';
+
+const regex = { tweet: /(?<=twitter.com\/(.*)\/status\/)\d*/i };
+
+// This is copy-pasted from src/timeline/TimelineItems.js
+// May make sense to separate into its own file depending on how often we end up needing this functionality
```
Comment (victoriamount) [inline src/components/EmbedSource.js:8]: Good call, I use it in 2 other places. I have a slightly different version I've been using that keeps that .com part but messes up when there's another . in the path, so it's not perfect yet

Comment (victoriamount) [reply src/components/EmbedSource.js:8]: Or maybe we don't like the .com and this is better!

**Review by victoriamount — APPROVED** [review]

File: src/components/EmbedSource.js (hunk 1 above)
Comment (mcolsen) [reply src/components/EmbedSource.js:8]: I have no opinion either way on including the TLD. My only concern is that currently it doesn't truncate any TLDs other than .com, so twitter.com becomes twitter but wikipedia.org will still be displayed as wikipedia.org.

**Review by starullo — APPROVED** [review]
Looks good to me! Thanks for doing that

**Pull request merged**